source/channel_bootstrap.c

```diff
@@ -92,6 +162,13 @@ void aws_client_bootstrap_clean_up(struct aws_client_bootstrap *bootstrap) {
 struct client_channel_data {
     struct aws_channel channel;
     struct aws_socket *socket;
+    struct aws_tls_connection_options tls_options;
+    aws_channel_on_protocol_negotiated_fn *on_protocol_negotiated;
+    aws_tls_on_data_read_fn *user_on_data_read;
+    aws_tls_on_negotiation_result_fn *user_on_negotiation_result;
+    aws_tls_on_error_fn *user_on_error;
+    void *tls_user_data;
+    bool use_tls;
```

Comment:
Is there any reason why this can't be divined from the presence of something in tls_options? Just feels slightly redundant